network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractLoadingLimits.java

```diff
@@ -100,4 +99,54 @@ public double getTemporaryLimitValue(int acceptableDuration) {
         LoadingLimits.TemporaryLimit tl = getTemporaryLimit(acceptableDuration);
         return tl != null ? tl.getValue() : Double.NaN;
     }
+
+    @Override
+    public T setTemporaryLimitValue(int acceptableDuration, double temporaryLimitValue) {
+        if (temporaryLimitValue < 0 || Double.isNaN(temporaryLimitValue)) {
+            throw new ValidationException(owner, "Temporary limit value must be a positive double");
+        }
+        TemporaryLimitAttributes identifiedLimit = attributes.getTemporaryLimits() == null ? null : attributes.getTemporaryLimits().get(acceptableDuration);
+        if (identifiedLimit == null) {
+            throw new ValidationException(owner, "No temporary limit found for the given acceptable duration");
+        }
```

Comment:
```TreeMap<Integer, TemporaryLimitAttributes> temporaryLimits = 
        new TreeMap<>(Optional.ofNullable(attributes.getTemporaryLimits())
                              .orElseThrow(() -> new ValidationException(owner, "No temporary limits available")));
        }
```

Reply:
well the error can not be reached beacause of 
```java
TemporaryLimitAttributes identifiedLimit = attributes.getTemporaryLimits() == null ? null : attributes.getTemporaryLimits().get(acceptableDuration);
        if (identifiedLimit == null) {
            throw new ValidationException(owner, "No temporary limit found for the given acceptable duration");
        }
```

Reply:
I think we can keep it like this to stick to the powsybl-core code